iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1923,13 +1926,22 @@ public Optional<ErrorMsg> isEligibleForCompaction(
   public List<Partition> getPartitions(org.apache.hadoop.hive.ql.metadata.Table table,
       Map<String, String> partitionSpec) throws SemanticException {
     return getPartitionNames(table, partitionSpec).stream()
-        .map(partName -> new DummyPartition(table, partName, partitionSpec)).collect(Collectors.toList());
+        .map(partName -> {
+          Map<String, String> partSpecMap = Maps.newLinkedHashMap();
+          Warehouse.makeSpecFromName(partSpecMap, new Path(partName), null);
+          return new DummyPartition(table, partName, partSpecMap);
+        }).collect(Collectors.toList());
   }
 
   @Override
   public Partition getPartition(org.apache.hadoop.hive.ql.metadata.Table table,
+      Map<String, String> partitionSpec, Configuration config) throws SemanticException {
```

Comment:
conf

Reply:
Conf was giving style check failures because it hades class member with the same name.
Maybe 'configuration'?

Reply:
not relevant anymore, passing RewritePolicy now as you suggested.